src/Compiler/Microsoft.AspNetCore.Razor.Language/src/SourceSpan.cs

```diff
@@ -96,6 +96,18 @@ public override string ToString()
             FilePath);
     }
 
+    internal SourceSpan With(int length, int endCharacterIndex)
```

Comment:
Why can't we calculate the new `length` given the new `endCharacterIndex`?

Reply:
`endCharacterIndex` is the index in the line, so if there are newlines then it won't be the same as just adding `length`. Aside: I would have liked to make this smarter, but the `SourceSpan` is a bit of a pain, as it has a `LineCount` property, but its just not used in some (all?) circumstances.